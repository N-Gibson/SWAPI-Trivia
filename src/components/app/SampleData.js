const SampleData = [{
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }, {
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }, {
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }, {
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }, {
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }, {
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }, {
    episode_id: 1,
    image: "https://vignette.wikia.nocookie.net/starwars/images/7/75/EPI_TPM_poster.png/revision/latest?cb=20130822171446",
    release_date: "1999-05-19",
    title: "The Phantom Menace",
    characters: [{
      films: ["The Phantom Menace"],
      homeworld: ["Mars", "234567890"],
      id: Date.now(),
      isFavorite: false,
      name: "Qui-Gon Jinn",
      species: "Human"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Naboo", "4500000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Roos Tarpals",
      species: "Gungan"
    }, {
      films: ["The Phantom Menace"],
      homeworld: ["Vulpter", "421000000"],
      id: Date.now(),
      isFavorite: false,
      name: "Dud Bolt",
      species: "Vulptereen"
    }]
  }]

  export default SampleData