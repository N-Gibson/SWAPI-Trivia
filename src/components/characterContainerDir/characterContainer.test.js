import React from 'react';
import { shallow } from 'enzyme';
import CharacterContainer from './characterContainer'
import SampleData from '../app/SampleData';

describe('CharacterContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CharacterContainer 
      characters={SampleData[0].characters} 
      favoriteCharacter={jest.fn()} 
      orderColor='blue' 
      name='name' 
      favQuote='favQuote' 
      rank='sith' 
      movieNumber='1' 
      moiveTitle='movietitle'
      scroll='scrolling text'
      />)
  })

  it.skip('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();    
  })

  it.skip('should update episode number', () => {

    console.log(wrapper.props().children[0].props)
    expect(wrapper.movieNumber).toEqual(1)
    expect(movieNumber).toEqual('I')
  })
})