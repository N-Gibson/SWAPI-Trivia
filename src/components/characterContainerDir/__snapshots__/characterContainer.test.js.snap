// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterContainer should match snapshot 1`] = `
<section
  className="container"
>
  <div
    className="fade"
  >
    <Nav
      favQuote="favQuote"
      name="name"
      orderColor="blue"
      rank="sith"
    />
  </div>
  <div
    className="star-wars-crawl"
  >
    <div
      className="crawl"
    >
      <div
        className="title"
      >
        <h3>
          Episode I
        </h3>
        <h2 />
      </div>
      scrolling text
    </div>
  </div>
  <div
    className="character-container"
  >
    <Character
      character={
        Object {
          "films": Array [
            "The Phantom Menace",
          ],
          "homeworld": Array [
            "Mars",
            "234567890",
          ],
          "id": 1571170247395,
          "isFavorite": false,
          "name": "Qui-Gon Jinn",
          "species": "Human",
        }
      }
      favoriteCharacter={[MockFunction]}
      orderColor="blue"
    />
    <Character
      character={
        Object {
          "films": Array [
            "The Phantom Menace",
          ],
          "homeworld": Array [
            "Naboo",
            "4500000000",
          ],
          "id": 1571170247395,
          "isFavorite": false,
          "name": "Roos Tarpals",
          "species": "Gungan",
        }
      }
      favoriteCharacter={[MockFunction]}
      orderColor="blue"
    />
    <Character
      character={
        Object {
          "films": Array [
            "The Phantom Menace",
          ],
          "homeworld": Array [
            "Vulpter",
            "421000000",
          ],
          "id": 1571170247395,
          "isFavorite": false,
          "name": "Dud Bolt",
          "species": "Vulptereen",
        }
      }
      favoriteCharacter={[MockFunction]}
      orderColor="blue"
    />
  </div>
</section>
`;
